src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +74,14 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p-> p.getAge() >= fromAge &&
+                        p.getSex() == Person.Sex.MAN &&
+                        p.getAge() <= maleToAge ||
+                        p.getAge() >= fromAge &&
+                        p.getSex() == Person.Sex.WOMAN &&
+                        p.getAge() <= femaleToAge)
+                .toList();
```

Comment:
Condition ```p.getAge() >= fromAge``` is duplicated here, please refactor this predicate to have common condition for ```fromAge``` and different for ```toAge``` based on ```p.getSex()```

Reply:
I've tried to refactor this by adding one more filter, but I'm not sure in this solution.